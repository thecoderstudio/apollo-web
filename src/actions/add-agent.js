const SELECT_OPERATING_SYSTEM = 'SELECT_OPERATING_SYSTEM';
const SELECT_ARCHITECTURE = 'SELECT_ARCHITECTURE';

function selectOperatingSystem(operatingSystem) {
  return {
    type: SELECT_OPERATING_SYSTEM,
    selectedOperatingSystem: operatingSystem
  };
}

function selectArchitecture(architecture) {
  return {
    type: SELECT_ARCHITECTURE,
    selectedArchitecture: architecture
  };
}

export {
  selectOperatingSystem,
  SELECT_OPERATING_SYSTEM,
  selectArchitecture,
  SELECT_ARCHITECTURE
};
