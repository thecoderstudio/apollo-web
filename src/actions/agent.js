export const LIST_AGENTS = 'LIST_AGENTS';

export const listAgents = (agents) => ({
  type: LIST_AGENTS,
  agents
});