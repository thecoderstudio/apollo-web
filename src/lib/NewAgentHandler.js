export default class NewAgentHandler {
  constructor() {
    this.supportedArch = ['amd64', 'arm64', 'arm'];
    this.supportedOS = ['linux', 'darwin', 'freebsd', 'openbsd'];
  };

  getDirectlyOnMachineCommand(os, arch, agentId, secret, host) {
    let command = `curl http://localhost:1970/agent/download?target_os=${os}&target_arch=${arch}`;
    return command += ` ${this.getExecuteCommand(agentId, secret, host)}`;
  };

  getExecuteCommand(agentId, secret, host) {
    return `chmod +x apollo-agent.bin && ./apollo-agent.bin --agent-id=${agentId} --secret=${secret} --host=${host}`;
  };
};