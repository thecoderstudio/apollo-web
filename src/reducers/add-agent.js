import { SHOW_ADD_AGENT_MODAL, CLOSE_ADD_AGENT_MODAL } from '../actions/add-agent';

export default function addAgentReducer(state = true, action) {
  switch (action.type) {
    case CLOSE_ADD_AGENT_MODAL:
      return false;
    case SHOW_ADD_AGENT_MODAL:
      return true;
    default: return state;
  }
}