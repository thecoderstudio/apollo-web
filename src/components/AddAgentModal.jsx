import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import axios from 'axios';
import Input from "./Input";
import Modal from './Modal';
import DescriptionButton from './buttons/DescriptionButton';
import Button from "./buttons/Button";
import DropDown from "./Drowdown";
import {Text} from "./Text";
import { closeAddAgentModal, selectArchitecture, selectOperatingSystem } from "../actions/add-agent";
import CopyToClipboard from "./CopyToClipboard";
import NewAgentHandler from "../lib/NewAgentHandler";


const TwoColumnGrid = styled.div`
  display: grid;
  grid-template-columns: [column-one] 50% [column-two] 50%;
  margin: 20px 0px 20px 0px;
`;

const ColumnOne = styled.div`
  grid-column: column-one;
  margin-right: 10px;
`;

const ColumnTwo = styled.div`
  grid-column: column-two;
  margin-left: 10px;
`;

const TextAndInputFieldWrapper = styled.div`
  display: grid;
  width: 100%;
  grid-template-columns: [text] 60% [dropdown] 40%; 
  height: 80px;
`;

const InputFieldWrapper = styled.div`
  grid-column: dropdown; 
`;

const TextWrapper = styled(Text)`
  grid-column: text;
  line-height: 75px;
`;

const StyledButton = styled(Button)`
  width: 200px;
  height: 50px;
  margin: auto;
  display: block;
`;

const DownloadBinaryButton = styled(StyledButton)`
  float: right;
`;

const ThreeRowDisplay = styled.div`
  display: grid;
  grid-template-rows: [description] 1fr [commands] 1fr [button] 1fr 
`;

const CommandWrapper = styled.div`
  grid-row: commands;
  margin-bottom: 20px;
`;

const CloseButton = styled(StyledButton)`
  grid-row: button;
  margin-top: 20px;
`;

const Description = styled(Text)`
  grid-row: description;
  margin: 20px 0px 20px 0px;
`;

const CreateAgentButton = styled(StyledButton)`
  margin-top: 20px;
`;

const StyledInput = styled(Input)`
  font-family: 'B612', sans-serif;
  font-weight: 600;
  width: 200px;
  padding: 15px;
  margin: 15px;
  background-color: ${props => props.theme.lightBlack};
  color: ${props => props.theme.white}; 
  border: 1px solid transparent;
   
  &:focus{
    outline: none;
    border: 1px solid ${props => props.theme.accent};  
  }
  
 
`;

class AddAgentModal extends React.PureComponent {
  constructor(props) {
    super(props);
    this.bindMethods()

    this.state = {
      renderFunction : this.renderQuestion,
      title: "Choose installation",
      agentName: ""
    };
    this.newAgentHandler = new NewAgentHandler();
  };

  bindMethods() {
    this.renderQuestion = this.renderQuestion.bind(this);
    this.setRenderFunction = this.setRenderFunction.bind(this);
    this.getStepOneComponents = this.getStepOneComponents.bind(this);
    this.selectStepOneDirectly = this.selectStepOneDirectly.bind(this);
    this.selectStepOneManual = this.selectStepOneManual.bind(this);
    this.renderDirectlyOnMachineStepOne = this.renderDirectlyOnMachineStepOne.bind(this);
    this.renderDirectlyOnMachineStepTwo = this.renderDirectlyOnMachineStepTwo.bind(this);
    this.renderManualUploadStepOne = this.renderManualUploadStepOne.bind(this);
    this.renderManualUploadStepTwo = this.renderManualUploadStepTwo.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.handleAgentNameChange = this.handleAgentNameChange.bind(this);
    this.createAgent = this.createAgent.bind(this);
    this.downloadBinary = this.downloadBinary.bind(this);
  }

  createAgent(renderFunctionCallback) {
    axios.post(
      `${process.env.APOLLO_HTTP_URL}agent`,
      { name : this.state.agentName },
      { withCredentials: true }
    )
      .then(response => {
        this.setRenderFunction(renderFunctionCallback);
      })
      .catch(error => {
        console.log(error)
      });
  }

  downloadBinary() {
    axios.get(
      `${process.env.APOLLO_HTTP_URL}agent/download`,
      {
        withCredentials: true,
        params: {
          target_os: this.props.selectedOperatingSystem,
          target_arch: this.props.selectedArchitecture
        },
        responseType: "blob",
      },
    )
      .then(response => {
        const url = window.URL.createObjectURL(new Blob([response.data]));
        const link = document.createElement('a');
        link.href = url;
        link.setAttribute('download', 'apollo-agent.binary');
        link.click();
      })
      .catch(error => {

      })
  }

  closeModal() {
    const { dispatch } = this.props;
    dispatch(closeAddAgentModal());
    this.setState({ renderFunction: this.renderQuestion });
  }

  setRenderFunction(renderFunction) {
    this.setState({ renderFunction: renderFunction});
  }

  selectStepOneDirectly(title) {
    this.setState({
      renderFunction: this.renderDirectlyOnMachineStepOne,
      title: title
    });
  };

  selectStepOneManual(title) {
    this.setState({
      renderFunction: this.renderManualUploadStepOne,
      title: title
    });
  };

  handleAgentNameChange(e) {
    this.setState({ agentName: e.target.value });
  }

  getStepOneComponents(onclick) {
    return(
      <div>
        <TextAndInputFieldWrapper>
          <TextWrapper>Agent name</TextWrapper>
          <InputFieldWrapper>
            <StyledInput placeholder="007" onChange={this.handleAgentNameChange} />
          </InputFieldWrapper>
        </TextAndInputFieldWrapper>
        <TextAndInputFieldWrapper>
          <TextWrapper>Operating system</TextWrapper>
          <InputFieldWrapper>
            <DropDown
              selected={this.props.selectedOperatingSystem}
              options={this.newAgentHandler.supportedOS}
              optionSelectedAction={selectOperatingSystem}
            />
          </InputFieldWrapper>
        </TextAndInputFieldWrapper>
        <TextAndInputFieldWrapper>
          <TextWrapper>Architecture</TextWrapper>
          <InputFieldWrapper>
            <DropDown
              selected={this.props.selectedArchitecture}
              options={this.newAgentHandler.supportedArch}
              optionSelectedAction={selectArchitecture}
            />
          </InputFieldWrapper>
        </TextAndInputFieldWrapper>
        <CreateAgentButton onClick={() => this.createAgent(onclick)}>
          Create agent
        </CreateAgentButton>
      </div>
    );
  };

  renderDirectlyOnMachineStepOne() {
    return this.getStepOneComponents(this.renderDirectlyOnMachineStepTwo);
  };

  getStepTwoComponents(command) {
   return(
      <ThreeRowDisplay>
        <Description>
          Copy and run the command on the target machine to install the client.
        </Description>
        <CommandWrapper>
          <CopyToClipboard text={command} />
        </CommandWrapper>
        <CloseButton onClick={this.closeModal}>Close</CloseButton>
      </ThreeRowDisplay>
    );
  }

  renderDirectlyOnMachineStepTwo() {
    return this.getStepTwoComponents("command");
  };

  renderManualUploadStepOne() {
    return this.getStepOneComponents(this.renderManualUploadStepTwo);
  };

  renderManualUploadStepTwo() {
    return(
      <div>
        <TwoColumnGrid>
          <ColumnOne>
            <Text>Download the binary and upload it to the target machine.</Text>
          </ColumnOne>
          <ColumnTwo>
            <DownloadBinaryButton onClick={this.downloadBinary}>Download binary</DownloadBinaryButton>
          </ColumnTwo>
        </TwoColumnGrid>
        {this.getStepTwoComponents("command")}
      </div>
    );
  };

  renderQuestion() {
    const directly = "Directly on target machine.";
    const manual = "Manual installation";
    return (
      <TwoColumnGrid>
        <ColumnOne>
          <DescriptionButton onClick={() => this.selectStepOneDirectly(directly)} title={directly}>
          You have the correct permissions to download and install the binary directly on the target machine.
          </DescriptionButton>
        </ColumnOne>
        <ColumnTwo>
          <DescriptionButton onClick={() => this.selectStepOneManual(manual)} title={manual}>
            You download, upload and install the binary yourself.
          </DescriptionButton>
        </ColumnTwo>
      </TwoColumnGrid>
    );
  };

  render() {
    return (
      <Modal visible={this.props.modalVisible} title={this.state.title}>
        {this.state.renderFunction()}
      </Modal>
    );
  };
}

export default connect(
  state => ({
    modalVisible: state.addAgent.modalVisible,
    selectedOperatingSystem: state.addAgent.selectedOperatingSystem,
    selectedArchitecture: state.addAgent.selectedArchitecture
  })
)(AddAgentModal)