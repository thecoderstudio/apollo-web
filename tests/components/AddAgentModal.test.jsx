import React from 'react';
import configureStore from 'redux-mock-store';
import waitForExpect from 'wait-for-expect';
import { Provider } from 'react-redux';
import axios from 'axios';
import renderer from 'react-test-renderer';
import AddAgentModal from '../../src/components/AddAgentModal';
import { closeAddAgentModal, } from "../../src/actions/add-agent";
import NewAgentHandler from '../../src/lib/NewAgentHandler';

const mockStore = configureStore([]);
jest.mock('axios');

function getComponent(store) {
  return renderer.create(
    <Provider store={store}>
      <AddAgentModal />
    </Provider>
  );
}

function getFinalPageComponent(store, buttonId) {
  const component = getComponent(store);
  const instance = component.root;
  instance.findByProps({ id: buttonId }).props.onClick();
  component.toJSON();
  instance.findByProps({ placeholder: '007' }).props.onChange({ target: {
    value: 'test'
  }});

  instance.findByProps({ loading: false }).props.onClick();
  return component;
}

describe('addAgentModal', () => {
  let store;
  let spy;

  beforeEach(() => {
    store = mockStore({
      addAgent: {
        modalVisible: true,
        selectedArchitecture: 'amd64',
        selectedOperatingSystem: 'linux'
      },
    });
    spy = jest.spyOn(store, 'dispatch');
  });

  it('renders correctly', () => {
    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('selects directly correctly', () => {
    const component = getComponent(store)
    const instance = component.root;

    instance.findByProps({ id: 'directlyButton' }).props.onClick();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('selects manual correctly', () => {
    const component = getComponent(store)
    const instance = component.root;
    instance.findByProps({ id: 'manualButton' }).props.onClick();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('closes modal correctly', async () => {
    const component = getComponent(store)
    const instance = component.root;
    instance.findByProps({ id: 'directlyButton' }).props.onClick();
    component.toJSON();
    instance.findByProps({ id: 'closeButton' }).props.onClick();
    expect(component.toJSON()).toMatchSnapshot();

    await waitForExpect(() => {
      expect(spy).toHaveBeenCalledWith(closeAddAgentModal());
    });
  });

  it('check for empty agent name', () => {
    const component = getComponent(store)
    const instance = component.root;
    instance.findByProps({ id: 'directlyButton' }).props.onClick();
    component.toJSON();
    instance.findByProps({ loading: false }).props.onClick();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('creates agent', async () => {
    axios.post.mockResolvedValue({
      status: 200,
      data: { id: 'test', oauth_client: { secret: 'test' } }
    });

    const component = getFinalPageComponent(store, 'directlyButton');
    await waitForExpect(() => {
      expect(component.toJSON()).toMatchSnapshot();
    })
  });

  it('creates agent incorrect name error', async () => {
    axios.post.mockRejectedValue({
      status: 400,
      response: { data: { name: 'error' } }
    });

    const component = getFinalPageComponent(store, 'directlyButton');

    await waitForExpect(async () => {
      expect(component.toJSON()).toMatchSnapshot();
      axios.post.mockRejectedValue({
        status: 400,
        response: { data: { somethingElse: 'error' } }
      });

      component.root.findByProps({ loading: false }).props.onClick();
      await waitForExpect(() => {
        expect(component.toJSON()).toMatchSnapshot();
      });
    });
  });

  it('calls download file correctly', async () => {
    axios.post.mockResolvedValue({
      status: 200,
      data: { id: 'test', oauth_client: { secret: 'test' } }
    });

    const spy = jest.spyOn(NewAgentHandler.prototype, 'downloadFile').mockImplementation(() => { return; });
    const component = getFinalPageComponent(store, 'manualButton');
    const instance = component.root;
    component.toJSON();

    axios.get.mockResolvedValue({
      status: 200,
      data: ''
    });

    await waitForExpect(() => {
      instance.findByProps({ id: 'downloadBinaryButton' }).props.onClick();
      expect(spy).toHaveBeenCalled();
    });
  });
});