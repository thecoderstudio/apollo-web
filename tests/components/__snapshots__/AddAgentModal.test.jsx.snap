// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAgentModal check for empty agent name 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Directly on target machine.
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Agent name
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <input
              className="sc-AxjAm sc-fzqBkg haYrtI"
              onChange={[Function]}
              placeholder="007"
            />
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Operating system
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                linux
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  darwin
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  freebsd
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  openbsd
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Architecture
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                amd64
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm64
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fznJRM jIWmxl"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzqPZZ eqyYUo"
            id="closeButton"
            onClick={[Function]}
          >
            Close
          </button>
          <div
            className="sc-fzoXWK gBiqkr"
          >
            <button
              className="sc-AxheI sc-fznxsB deHuBN"
              onClick={[Function]}
            >
              Create agent
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal closes modal correctly 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Directly on target machine.
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div
        className="sc-fznJRM jIWmxl"
      >
        <div
          className="sc-fznWqX ULKNC"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzoLsD bCkLQg"
            onClick={[Function]}
          >
            <h3
              className="sc-fzpans jXuSvn"
            >
              Directly on target machine.
            </h3>
            <span
              className="sc-fzozJi sc-fzplWN dJDgft"
            >
              You have the correct permissions to download the binary directly on the target machine.
            </span>
          </button>
        </div>
        <div
          className="sc-fzoiQi jXGlMn"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzoLsD bCkLQg"
            onClick={[Function]}
          >
            <h3
              className="sc-fzpans jXuSvn"
            >
              Manual upload
            </h3>
            <span
              className="sc-fzozJi sc-fzplWN dJDgft"
            >
              You download and upload the binary yourself.
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal creates agent 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Directly on target machine.
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div
        className="sc-fznxKY hOhmQT"
      >
        <span
          className="sc-AxmLO sc-fzolEj dTXSgD"
        >
          Copy and run the command on the target machine to download run the client.
        </span>
        <div
          className="sc-fznMAR glbjV"
        >
          <div
            className="sc-fzqNJr lglQoL"
          >
            <div
              className="sc-fzoyAV dZsvGk"
            >
              curl http://localhost:1970/agent/download?target_os=linux&target_arch=amd64 && chmod +x apollo-agent.bin && ./apollo-agent.bin --agent-id=test --secret=test --host=localhost:1970
            </div>
            <button
              className="sc-AxheI sc-fzoLag jgDeBs"
              id="copyToClipboardButton"
              onClick={[Function]}
            >
              <img
                className="sc-fzoXzr hlhLlT"
                src={[Function]}
              />
            </button>
          </div>
        </div>
        <button
          className="sc-AxheI sc-fzoNJl sc-fznWOq gzFGsa"
          id="closeButton"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal creates agent incorrect name error 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Directly on target machine.
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Agent name
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <input
              className="sc-AxjAm sc-fzqBkg haYrtI"
              onChange={[Function]}
              placeholder="007"
            />
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Operating system
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                linux
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  darwin
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  freebsd
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  openbsd
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Architecture
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                amd64
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm64
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fznJRM jIWmxl"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzqPZZ eqyYUo"
            id="closeButton"
            onClick={[Function]}
          >
            Close
          </button>
          <div
            className="sc-fzoXWK gBiqkr"
          >
            <button
              className="sc-AxheI sc-fznxsB deHuBN"
              onClick={[Function]}
            >
              Create agent
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal creates agent incorrect name error 2`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Directly on target machine.
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Agent name
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <input
              className="sc-AxjAm sc-fzqBkg haYrtI"
              onChange={[Function]}
              placeholder="007"
            />
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Operating system
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                linux
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  darwin
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  freebsd
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  openbsd
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Architecture
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                amd64
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm64
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fznJRM jIWmxl"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzqPZZ eqyYUo"
            id="closeButton"
            onClick={[Function]}
          >
            Close
          </button>
          <div
            className="sc-fzoXWK gBiqkr"
          >
            <button
              className="sc-AxheI sc-fznxsB deHuBN"
              onClick={[Function]}
            >
              Create agent
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal renders correctly 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Choose installation
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div
        className="sc-fznJRM jIWmxl"
      >
        <div
          className="sc-fznWqX ULKNC"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzoLsD bCkLQg"
            onClick={[Function]}
          >
            <h3
              className="sc-fzpans jXuSvn"
            >
              Directly on target machine.
            </h3>
            <span
              className="sc-fzozJi sc-fzplWN dJDgft"
            >
              You have the correct permissions to download the binary directly on the target machine.
            </span>
          </button>
        </div>
        <div
          className="sc-fzoiQi jXGlMn"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzoLsD bCkLQg"
            onClick={[Function]}
          >
            <h3
              className="sc-fzpans jXuSvn"
            >
              Manual upload
            </h3>
            <span
              className="sc-fzozJi sc-fzplWN dJDgft"
            >
              You download and upload the binary yourself.
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal selects directly correctly 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Directly on target machine.
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Agent name
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <input
              className="sc-AxjAm sc-fzqBkg dAOHwa"
              onChange={[Function]}
              placeholder="007"
            />
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Operating system
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                linux
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  darwin
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  freebsd
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  openbsd
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Architecture
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                amd64
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm64
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fznJRM jIWmxl"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzqPZZ eqyYUo"
            id="closeButton"
            onClick={[Function]}
          >
            Close
          </button>
          <div
            className="sc-fzoXWK gBiqkr"
          >
            <button
              className="sc-AxheI sc-fznxsB deHuBN"
              onClick={[Function]}
            >
              Create agent
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`addAgentModal selects manual correctly 1`] = `
<div
  className="sc-AxiKw eqPaRe"
>
  <div
    className="sc-AxirZ sc-AxhCb gWcjrs"
  >
    <h2
      className="sc-AxhUy kyqeXI"
    >
      Manual upload
    </h2>
    <div
      className="sc-AxgMl qesma"
    >
      <div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Agent name
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <input
              className="sc-AxjAm sc-fzqBkg dAOHwa"
              onChange={[Function]}
              placeholder="007"
            />
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Operating system
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                linux
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  darwin
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  freebsd
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  openbsd
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqARJ gIhxgi"
        >
          <span
            className="sc-AxmLO sc-fzoyTs jjJyjZ"
          >
            Architecture
          </span>
          <div
            className="sc-fzqNqU bDXkCM"
          >
            <div
              className="sc-fznyAO crUbpk"
            >
              <div
                className="sc-fznKkj cfTTXb"
                id="dropdown"
                onClick={[Function]}
              >
                amd64
                <div
                  className="sc-fzqBZW jnzxYq"
                />
              </div>
              <ul
                className="sc-fznZeY xMdrF"
              >
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm64
                </li>
                <li
                  className="sc-fzokOt gDZuMP"
                  onClick={[Function]}
                >
                  arm
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="sc-fznJRM jIWmxl"
        >
          <button
            className="sc-AxheI sc-Axmtr sc-fzqPZZ eqyYUo"
            id="closeButton"
            onClick={[Function]}
          >
            Close
          </button>
          <div
            className="sc-fzoXWK gBiqkr"
          >
            <button
              className="sc-AxheI sc-fznxsB deHuBN"
              onClick={[Function]}
            >
              Create agent
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
